File: tests/java/org/python/tests/identity/IdentityTest.java
package org.python.tests.identity;

import junit.framework.TestCase;

import org.python.core.PyString;
import org.python.core.PyStringMap;
import org.python.core.PySystemState;
import org.python.util.PythonInterpreter;

public class IdentityTest extends TestCase {

    private PythonInterpreter interp;

    @Override
    protected void setUp() throws Exception {
        PySystemState sys = new PySystemState();
        interp = new PythonInterpreter(new PyStringMap(), sys);
    }

    public void testReadonly() {
        interp.execfile("tests/python/identity_test.py");
    }
}

File: tests/python/identity_test.py
from org.python.tests.identity import IdentityObject

class PyIdentityObject (IdentityObject):
    pass

x = IdentityObject()
y = x.getThis()
        
a = PyIdentityObject()
b = a.getThis()
        
assert x is y
assert not(x is not y)
assert (x is y)  !=  (x is not y)
assert a is b
assert not(a is not b)
assert (a is b)  !=  (a is not b)

